dapp-core/src/main/kotlin/jp/co/soramitsu/dapp/service/ContractsRepositoryMonitor.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jp.co.soramitsu.dapp.service
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.google.gson.JsonParser
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import jp.co.soramitsu.dapp.listener.ReliableIrohaChainListener
+import jp.co.soramitsu.iroha.java.QueryAPI
+import mu.KLogging
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import java.util.concurrent.Executors
+
+@Component
+class ContractsRepositoryMonitor(
+    @Autowired
+    private val chainListener: ReliableIrohaChainListener,
+    @Autowired
+    private val queryAPI: QueryAPI,
+    @Autowired
+    private val dAppAccountId: String,
```

Comment:
How does it work? Three `String` beans and no conflicts?

Reply:
it just works lol